src/serializer.rs

```diff
@@ -105,9 +105,9 @@ impl SdPayload {
         writer.write_u8(self.flags)?;
         writer.write_u24::<BigEndian>(0x000000)?; // reserved
 
-        writer.write_u32::<BigEndian>(self.entries_len() as u32)?;
+        writer.write_u32::<BigEndian>(self.entries_length() as u32)?;
         for entry in &self.entries {
-            let entry_type = entry.r#type();
+            let entry_type = entry.get_type();
```

Comment:
In Rust the prefix `get_` is very uncommon because literally every expression is a "get".

Reply:
resolved by impl From<&SdEntry> for u8